### PR TITLE
[appservice-kube] `az functionapp create`: Make `--storage-account` optional for Kubernetes function apps 

### PR DESCRIPTION
**Pull request by StrawnSC**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
appservice-kube

**Comment by panchagnula** [comment]
FYI @ lpapudippu

**Review by wangzelin007 — APPROVED** [review]

**Comment by wangzelin007** [comment]
@[StrawnSC](https://github.com/StrawnSC),
Please fix the integration test error.

**Comment by panchagnula** [comment]
@StrawnSC added do not merge label for now, since there is some confusion, that feature team wants to test this with private whl, before releasing. you are on the group chat as well, lets merge this once we get conformation from them. Thanks!

**Comment by StrawnSC** [comment]
@zhoxing-ms could we get this released soon as well? Thanks!

File: src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py
Comment (zhoxing-ms) [inline src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py:21]: Why do you mark this test as `@live_only()`?

Comment (StrawnSC) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: This test passes locally (run live or as a recording), but fails in the CI with: `vcr.errors.CannotOverwriteExistingCassetteException`. I just pushed the recording for this test and removed the live_only and now the tests are failing

Comment (zhoxing-ms) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: > ERROR: Can't overwrite existing cassette ('/home/vsts/work/1/s/src/appservice-kube/azext_appservice_kube/tests/latest/recordings/test_linux_webapp_quick_create_kube.yaml') in your current record mode ('once').
No match for the request (<Request (POST) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.Web/validate?api-version=2021-03-01>) was found.
Found 6 similar requests with 1 different matcher(s) :

> 1 - (<Request (POST) https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Web/checknameavailability?api-version=2021-03-01>).
Matchers succeeded : ['method', 'scheme', 'host', 'port', '_custom_request_query_matcher']
Matchers failed :
path - assertion failure :
/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/clitest.rg000001/providers/Microsoft.Web/validate != /subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Web/checknameavailability

This problem is caused by the fact that a rest request for validation is not recorded in the yaml file

Comment (zhoxing-ms) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: @StrawnSC I guess this may be because the code used in the local tests is not up to date. 
Could you pull the latest code from CLI main repo and CLI extension repo to the local, and then try to re-record the yaml file?

Comment (StrawnSC) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: @zhoxing-ms it's still failing even when I use the latest code for the main CLI (and rerun `azdev setup`)

Comment (StrawnSC) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: This testing issue seems unrelated to this change: the change was to `az functionapp create` whereas the test that passes locally but fails in the CI is testing `az webapp create`. Also, we've been getting requests from the function apps team to release this for a while. Can I go ahead and mark the test as `live_only` for now so we can release this?

Comment (zhoxing-ms) [reply src/appservice-kube/azext_appservice_kube/tests/latest/test_appservice_kube_scenario.py]: OK, if you are sure that this problem does not affect the feature of this PR, you can mark the test as `live_only` to avoid blocking the merging of PR. You can solve the test problems in the follow-up PR

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
